docs/en/advanced/Tiered-Locality.md

```diff
@@ -6,59 +6,102 @@ group: Advanced
+To configure tiered locality, follow these steps:
+
+1. Write a script named `alluxio-locality.sh`:
+   ```
+   #!/bin/bash
+  
+   node=$(hostname)
+   # Ask EC2 which availability zone we're in
+   availability_zone=$(curl -s http://169.254.169.254/latest/meta-data/placement/availability-zone)
+   
+   echo "node=${node},availability_zone=${availability_zone}"
+   ```
+  
+1. Make the script executable with `chmod +x alluxio-locality.sh`.
+
+1. Include the script on the classpath of your applications and Alluxio servers. For servers,
+   you can put the file in the `conf` directory.
+  
+1. On the Alluxio master(s), set `alluxio.locality.order=node,availability_zone` to define the order
+   of locality tiers.
+  
+1. Restart Alluxio servers to pick up the configuration changes.
+
+To verify that the configuration is working, check the master, worker, and application logs. You
+should see a line like
+
+```
+INFO  TieredIdentityFactory - Initialized tiered identity TieredIdentity(node=ip-xx-xx-xx-xx, availability_zone=us-east-1)
+```
+
+If you're having problems, try running your locality script directly to check its output, and make sure 
+it is executable by the user running the Alluxio servers.
+
+## Advanced
 
-## Configuration
+### Custom locality tiers
+
+By default, Alluxio has two locality tiers: `node` and `rack`. Users may customize the
+set of locality tiers to take advantage of more advanced topologies. To change the set
+of tiers available, set `alluxio.locality.order` on the master and restart the cluster.
+The order should go from most specific to least specific. For example, to add
+availability zone locality to a cluster, set
+
+```
+alluxio.locality.order=node,rack,availability_zone
+```
 
 If the user does nothing to provide tiered identity info, each entity will
 perform a localhost lookup to set its node-level identity info. If other locality tiers
 are left unset, they will not be used to inform locality decisions. To set
 the value for a locality tier, set the configuration property
 
-```
-alluxio.locality.[tiername]=...
-```
+### Custom locality script name
 
-See the [Configuration-Settings](Configuration-Settings.html) page for details on how
-to set configuration properties.
-
-It is also possible to configure tiered identity info via script. By default Alluxio searches
-the classpath for a script named `alluxio-locality.sh`. You can override this name by setting
+By default Alluxio searches the classpath for a script named `alluxio-locality.sh`. You can override this name by setting
 
 ```
-alluxio.locality.script=/path/to/script
+alluxio.locality.script=locality_script_name
 ```
 
-The script should be executable and output a comma-separated list of `tierName=...`
-pairs. Here is an example script for reference:
+### Set locality via configuration properties
 
-```bash
-#!/bin/bash
+In some situations, it can be more convenient to set locality information with Alluxio configuration properties
```

Comment:
Is there an example of such situation? Which way is recommended?

Reply:
Recommended using the `alluxio-locality.sh` script unless the user doesn't have access to the classpath